ADAL/src/telemetry/ADAggregatedDispatcher.m

```diff
@@ -40,40 +40,57 @@ - (id)initWithDispatcher:(id<ADDispatcher>)dispatcher
     return self;
 }
 
-- (void)flush
+- (void)flush:(NSString*)requestId
 {
     [_dispatchLock lock]; //avoid access conflict when manipulating _objectsToBeDispatched
-    NSMutableDictionary* objectsToBeDispatched = _objectsToBeDispatched;
-    _objectsToBeDispatched = [NSMutableDictionary new];
+    NSArray* eventsToBeDispatched = [_objectsToBeDispatched objectForKey:requestId];
+    [_objectsToBeDispatched removeObjectForKey:requestId];
     [_dispatchLock unlock];
     
     // Integrate events of a particular request id into one single event
-    for (NSString* requestId in objectsToBeDispatched)
+    NSMutableArray* aggregatedEvent = [NSMutableArray new];
+    SAFE_ARC_AUTORELEASE(aggregatedEvent);
+    
+    for (id<ADTelemetryEventInterface> event in eventsToBeDispatched)
     {
-        NSArray* events = [objectsToBeDispatched objectForKey:requestId];
-        NSMutableArray* aggregatedEvent = [NSMutableArray new];
+        NSArray* properties = [event getProperties];
+        
+        NSInteger propertiesToSkip = 0;
+        // default properties are duplicate for all events,
+        // so they are skipped from 2nd event onwards
+        if (event != [eventsToBeDispatched objectAtIndex:0])
+        {
+            propertiesToSkip = [event getDefaultPropertyCount];
+        }
         
-        for (id<ADEventInterface> event in events)
```

Comment:
Any reason for not using the for ( obj in collection) syntax? It's a little bit faster.